Sources/MapboxSearch/PublicAPI/Use Cases/Address Autofill/Models/Suggestion/CoreAddress+AddressComponents.swift

```diff
@@ -0,0 +1,33 @@
+// Copyright © 2023 Mapbox. All rights reserved.
+
+import Foundation
+
+extension CoreAddress {
+    enum AutofillParsingError: Error {
+        case emptyAddressComponents
+    }
+
+    func toAutofillComponents() throws -> NonEmptyArray<AddressAutofill.AddressComponent> {
+        var components: [AddressAutofill.AddressComponent] = []
+
+        houseNumber.map { components.append(.init(kind: .houseNumber, value: $0)) }
+        street.map { components.append(.init(kind: .street, value: $0)) }
+        neighborhood.map { components.append(.init(kind: .neighborhood, value: $0)) }
+        locality.map { components.append(.init(kind: .locality, value: $0)) }
+        postcode.map { components.append(.init(kind: .postcode, value: $0)) }
+        place.map { components.append(.init(kind: .place, value: $0)) }
+        district.map { components.append(.init(kind: .district, value: $0)) }
+        region.map { components.append(.init(kind: .region, value: $0)) }
+        country.map { components.append(.init(kind: .country, value: $0)) }
```

Comment:
Can we rewrite this as a for loop, using `CaseIterable`?
Right now there is a chance we can forget to update this function after adding a new case to AddressComponent.Kind

Reply:
We would still have to line up each `components` array member with the field from `CoreAddress` to assign the data to the right enum case. This is built [following the pattern from AddressAutofill.Suggestion+SearchResult](https://github.com/mapbox/mapbox-search-ios/blob/main/Sources/MapboxSearch/PublicAPI/Use%20Cases/Address%20Autofill/Models/Suggestion/AddressAutofill.Suggestion%2BSearchResult.swift#L41).